applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,503 @@
+     * @param MoveMeshFlag The flag that allows to move the mesh
+     */
+    explicit GeoMechanicNewtonRaphsonStrategyLinearElasticDynamic(
+        ModelPart&                                 rModelPart,
+        typename TSchemeType::Pointer              pScheme,
+        typename TLinearSolver::Pointer            pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer    pNewBuilderAndSolver,
+        Parameters&                                rParameters,
+        int                                        MaxIterations      = 30,
+        bool                                       CalculateReactions = false,
+        bool                                       MoveMeshFlag       = false)
+        : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(
+              rModelPart, pScheme, pNewConvergenceCriteria, pNewBuilderAndSolver, MaxIterations, CalculateReactions, false, MoveMeshFlag)
+    {
+        // new constructor
+    }
+
+    /**
+     * @brief Initialization of member variables and prior operations
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY;
+
+        BaseType::Initialize();
+
+        // Note that FindNeighbourElementsOfConditionsProcess and DeactivateConditionsOnInactiveElements are required to be perfomed before initializing the System and State
+        // this means that these operations are done twice in the GeomechanicsSolver in python
+        auto find_neighbour_elements_of_conditions_process =
+            FindNeighbourElementsOfConditionsProcess(BaseType::GetModelPart());
+        find_neighbour_elements_of_conditions_process.Execute();
+
+        auto deactivate_conditions_on_inactive_elements_process =
+            DeactivateConditionsOnInactiveElements(BaseType::GetModelPart());
+        deactivate_conditions_on_inactive_elements_process.Execute();
+
+        if (!BaseType::mStiffnessMatrixIsBuilt)
+            // initialize the system matrices and the initial second derivative
+            this->InititalizeSystemAndState();
+
+        KRATOS_CATCH("");
+    }
+
+    void Predict() override
+    {
+        KRATOS_TRY
+        // OPERATIONS THAT SHOULD BE DONE ONCE - internal check to avoid repetitions
+        // if the operations needed were already performed this does nothing
+        if (BaseType::mInitializeWasPerformed == false) this->Initialize();
+
+        TSystemMatrixType& rA  = *BaseType::mpA;
+        TSystemVectorType& rDx = *BaseType::mpDx;
+        TSystemVectorType& rb  = *BaseType::mpb;
+
+        DofsArrayType& r_dof_set = BaseType::GetBuilderAndSolver()->GetDofSet();
+
+        BaseType::GetScheme()->Predict(BaseType::GetModelPart(), r_dof_set, rA, rDx, rb);
+
+        // Note that constraints are not applied in this predict, nor is an update performed
+
+        // Move the mesh if needed
+        if (BaseType::MoveMeshFlag() == true) BaseType::MoveMesh();
```

Comment:
This reads like Fortran ;-) Please use common C++:
```suggestion
        if (BaseType::MoveMeshFlag()) BaseType::MoveMesh();
```